appimage/private/runfiles.bzl

```diff
@@ -87,15 +87,16 @@ def collect_runfiles_info(ctx):
     Returns:
         struct with infos about files needed by app.
     """
-    runfiles_list = _default_runfiles(ctx.attr.binary).to_list()
-    file_map = {_final_file_path(ctx, f): f for f in runfiles_list}
+    runfiles_list = depset(ctx.files.data, transitive = [_default_runfiles(ctx.attr.binary)] + [_default_runfiles(d) for d in ctx.attr.data]).to_list()
+    symlinks_list = depset(transitive = [_default_symlinks(ctx.attr.binary)] + [_default_symlinks(d) for d in ctx.attr.data]).to_list()
+    emptyfiles_list = depset(transitive = [_default_emptyfiles(ctx.attr.binary)] + [_default_emptyfiles(d) for d in ctx.attr.data]).to_list()
```

Comment:
So the depset is just for de-duplication at least for symlinks and emptyfiles? Might be nice to add an explaining comment.